java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java

```diff
@@ -1142,8 +1142,14 @@ public MantaObjectResponse put(final String rawPath,
         Validate.notNull(source, "Input stream must not be null");
         final String path = formatPath(rawPath);
 
-        final ContentType contentType = ContentTypeLookup.findOrDefaultContentType(headers,
-                ContentType.APPLICATION_OCTET_STREAM);
+        final ContentType contentType;
+
+        if (BooleanUtils.isTrue(config.isContentTypeDetectionEnabled())) {
```

Comment:
It just struck me. Couldn't we do a static private method for this within `MantaClient`, this would somewhat cut down on the code duplication. What do you think?

Reply:
I wholeheartedly agree.